src/Cli/dotnet/commands/dotnet-sdk/check/Program.cs

```diff
@@ -29,18 +30,31 @@ public class SdkCheckCommand : CommandBase
 
         public override int Execute()
         {
-            var dotnetPath = EnvironmentProvider.GetDotnetExeDirectory();
-            var productCollection = _productCollectionProvider.GetProductCollection();
-            var environmentInfo = _netBundleProvider.GetDotnetEnvironmentInfo(dotnetPath);
-            var sdkFormatter = new SdkOutputWriter(environmentInfo.SdkInfo, productCollection, _productCollectionProvider, _reporter);
-            var runtimeFormatter = new RuntimeOutputWriter(environmentInfo.RuntimeInfo, productCollection, _productCollectionProvider, _reporter);
-
-            sdkFormatter.PrintSdkInfo();
-            _reporter.WriteLine();
-            runtimeFormatter.PrintRuntimeInfo();
-            _reporter.WriteLine();
-            _reporter.WriteLine(LocalizableStrings.CommandFooter);
-            _reporter.WriteLine();
+            try
+            {
+                var dotnetPath = EnvironmentProvider.GetDotnetExeDirectory();
+                var productCollection = _productCollectionProvider.GetProductCollection();
+                var environmentInfo = _netBundleProvider.GetDotnetEnvironmentInfo(dotnetPath);
+                var sdkFormatter = new SdkOutputWriter(environmentInfo.SdkInfo, productCollection, _productCollectionProvider, _reporter);
+                var runtimeFormatter = new RuntimeOutputWriter(environmentInfo.RuntimeInfo, productCollection, _productCollectionProvider, _reporter);
+
+                sdkFormatter.PrintSdkInfo();
+                _reporter.WriteLine();
+                runtimeFormatter.PrintRuntimeInfo();
+                _reporter.WriteLine();
+                _reporter.WriteLine(LocalizableStrings.CommandFooter);
+                _reporter.WriteLine();
+            }
+            catch (HostFxrRuntimePropertyNotSetException)
+            {
+                _reporter.WriteLine(string.Format(LocalizableStrings.RuntimePropertyNotFound));
+                return -1;
+            }
+            catch (HostFxrNotFoundException e)
+            {
+                _reporter.WriteLine(string.Format(LocalizableStrings.HostFxrCouldNotBeLoaded, e.Message));
+                return -1;
+            }
```

Comment:
Non-blocking (feel free to merge and fix this later):

Cleaner solution would probably be to rethrow `GracefulException` which seems to be designed to provide good error messages for known cases.

Set the `isUserError = false` and then provide a good error message.
The command execution logic will catch the exception and report it to the user - https://github.com/dotnet/sdk/blob/a6e14e80ca216d6a394387a628f0e3abbadd6f28/src/Cli/dotnet/DotNetTopLevelCommandBase.cs#L48.